crates/pathfinder/src/rpc/v02/method/get_class.rs

```diff
@@ -0,0 +1,327 @@
+use crate::core::{BlockId, ClassHash};
+use crate::rpc::v02::types::ContractClass;
+use crate::rpc::v02::RpcContext;
+
+use anyhow::Context;
+use rusqlite::OptionalExtension;
+
+crate::rpc::error::generate_rpc_error_subset!(GetClassError: BlockNotFound, ClassHashNotFound);
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct GetClassInput {
+    block_id: BlockId,
+    class_hash: ClassHash,
+}
+
+pub async fn get_class(
+    context: RpcContext,
+    input: GetClassInput,
+) -> Result<ContractClass, GetClassError> {
+    let block = match input.block_id {
+        BlockId::Pending => {
+            if is_pending_class(&context.pending_data, input.class_hash).await {
+                BlockId::Pending
+            } else {
+                BlockId::Latest
+            }
+        }
+        other => other,
+    };
+
+    let span = tracing::Span::current();
+    let jh = tokio::task::spawn_blocking(move || -> Result<ContractClass, GetClassError> {
+        let _g = span.enter();
+        let mut db = context
+            .storage
+            .connection()
+            .context("Opening database connection")?;
+        let tx = db.transaction().context("Creating database transaction")?;
+
+        let definition = match block {
+            BlockId::Pending => read_pending(&tx, input.class_hash),
+            BlockId::Number(number) => read_at_number(&tx, input.class_hash, number),
+            BlockId::Hash(hash) => read_at_hash(&tx, input.class_hash, hash),
+            BlockId::Latest => read_latest(&tx, input.class_hash),
```

Comment:
I tried a few times to get what looks like common code and sql statements together more, but I think this actually ended up more readible. I also failed to get something more factored.